src/Func.cpp

```diff
@@ -2908,11 +2908,17 @@ void Func::infer_input_bounds(int x_size, int y_size, int z_size, int w_size,
         // the halide_buffer_t fields. (We can't use crop because it explicitly
         // disallows expanding the fields in this unsafe manner.)
         Buffer<> im = Buffer<>::make_scalar(func.output_types()[i]);
```

Comment:
I thought the whole point of this code was that the host pointers on the output buffers need to be non-null. Otherwise it could be substantially less clever here - we could just make bounds query buffers. Maybe that's not the case?

Reply:
Assuming this is passing tests, maybe at some point in time the outputs needed to be non-null in order to do bounds inference on the inputs.

Reply:
> I thought the whole point of this code

That is an excellent question, but I don't know the answer. FWIW, test_correctness seems to pass just fine like this. AFAICT this is only used when we are going to infer the *input* bounds, so the only thing the outputs are even needed for are to provide dimensions.

Reply:
It's passing all the tests.

Reply:
In that case we can nuke the entire loop below and just say:
```
Buffer<> im(func.output_types()[i], nullptr, sizes);
```
for an appropriate `vector<int>` of sizes.